### PR TITLE
feat: introduce TGeoDetectorElementSplitter & splitted detector element constructors

### PR DESCRIPTION
**Pull request by asalzburger**
This PR introduces a new interface in the `TGeo` plugin that allows splitting of `TGeoDetectorElement` objects and adds a necessary constructor for this to the `TGeoDetectorElement`.

This will be needed for the ALICE o2 studies (generic splitter from disk to trapezoids), but also for the TGeo version of the ATLAS ITk, where dedicated splitters need to be written.

Tagging some interested parties: @paulgessinger @niermann999 @noemina 

The splitting has to be activated with a switch:

```shell
--geo-tgeo-cyl-disc-split
```

And then, consequently, all volumes need to define how they wanna perform the splitting, e.g. for no splitting:

```shell
--geo-tgeo-cyl-nz-segs -1
--geo-tgeo-cyl-nphi-segs -1
--geo-tgeo-disc-nr-segs -1
--geo-tgeo-disc-nphi-segs -1
```

versus splitting:

```shell
--geo-tgeo-cyl-nz-segs 6
--geo-tgeo-cyl-nphi-segs 32
--geo-tgeo-disc-nr-segs 6
--geo-tgeo-disc-nphi-segs 32
```

### CODE REVIEW

**Comment by asalzburger** [comment]
Splitting for `DiscSurface` layers is done:

Screenshot 2021-07-12 at 16 43 47

This is an example of `6` radial splits and `32` phi segments.

**Comment by asalzburger** [comment]
Screenshot 2021-07-15 at 13 52 18

Splitting for Cylinders enabled.

File: Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  /// Constructor
+  ///
+  /// @param cfg the configuration struct
+  /// @param logger the logging object
+  TGeoCylinderDiscSplitter(
+      const Config& cfg,
+      std::unique_ptr<const Acts::Logger> logger = Acts::getDefaultLogger(
+          "TGeoCylinderDiscSplitter", Acts::Logging::INFO));
+
+  virtual ~TGeoCylinderDiscSplitter() = default;
```
Comment (niermann999) [inline Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp:50]: Shouldn't this better go to the Interface?

Comment (asalzburger) [reply Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp:51]: I think because it's only purely virtual it's fine, did not complain.

Comment (niermann999) [reply Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp:51]: Ok, maybe I'm mistaken, but is the destrcutor of the splitter called through an interface pointer when the interface has no virtual destructor?

Comment (asalzburger) [reply Plugins/TGeo/include/Acts/Plugins/TGeo/TGeoCylinderDiscSplitter.hpp:51]: I've included it now.

**Review by niermann999 — APPROVED** [review]

**Pull request merged**